WiserTaskScheduler/WiserTaskScheduler/Modules/GenerateFiles/Services/GenerateFileService.cs

```diff
@@ -141,6 +149,40 @@ private async Task<JObject> GenerateFile(GenerateFileModel generateFile, List<in
                     Html = body
                 };
                 pdfFile = await htmlToPdfConverterService.ConvertHtmlStringToPdfAsync(pdfSettings);
+                
+                // Merge PDF's to generated PDF
+                if (generateFile.Body.MergePdfs.Where(p => !String.IsNullOrEmpty(p.WiserItemId)).ToArray().Length > 0)
+                {
+                    // Make stream of byte array
+                    var stream = new MemoryStream(pdfFile.FileContents);
+                    
+                    //  Create the merge result PDF document
+                    var mergeResultPdfDocument = new Document(stream);
+
+                    // Automatically close the merged documents when the document resulted after merge is closed
+                    mergeResultPdfDocument.AutoCloseAppendedDocs = true;
+
+                    // Set license key received after purchase to use the converter in licensed mode
+                    // Leave it not set to use the converter in demo mode
+                    mergeResultPdfDocument.LicenseKey = GclSettings.Current.EvoPdfLicenseKey;
+
+                    foreach (var pdf in generateFile.Body.MergePdfs)
+                    {
+                        if (String.IsNullOrEmpty(pdf.WiserItemId))
+                        {
+                            continue;
+                        }
+
+                        // Create a scope to get the string wiserItems service. It's needed to get the template
+                        var wiserItemsService = scope.ServiceProvider.GetRequiredService<IWiserItemsService>();
```

Comment:
Why do you do this inside the foreach? It's much more efficient to get the WiserItemsService once before the foreach and then use that every time inside the foreach.